src/ast/mod.rs

```diff
@@ -99,7 +100,30 @@ where
             GenericNCommand::Sort(span, name, params) => {
                 GenericCommand::Sort(span.clone(), *name, params.clone())
             }
-            GenericNCommand::Function(f) => GenericCommand::Function(f.clone()),
+            // This is awkward for the three subtypes change
+            GenericNCommand::Function(f) => match f.subtype {
+                FunctionSubtype::Constructor => GenericCommand::Constructor {
+                    span: f.span.clone(),
+                    name: f.name,
+                    schema: f.schema.clone(),
+                    cost: f.cost,
+                    unextractable: f.unextractable,
+                },
+                FunctionSubtype::Relation => GenericCommand::Relation {
+                    span: f.span.clone(),
+                    constructor: f.name,
+                    inputs: f.schema.input.clone(),
+                },
+                FunctionSubtype::Custom => GenericCommand::Function {
```

Comment:
Why call this "Custom"? Why not name it "Function"? Are there other non-function things this could be?

Reply:
We decided to call it "custom function" to distinguish it from "function," which contains all three subtypes.

Reply:
ah ok